src/main/java/by/exadel/internship/config/JpaAuditingConfiguration.java

```diff
@@ -0,0 +1,23 @@
+package by.exadel.internship.config;
+
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.domain.AuditorAware;
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Optional;
+
+@Configuration
+@EnableJpaAuditing(auditorAwareRef = "auditorProvider")
+public class JpaAuditingConfiguration {
+
+    @Bean
+    public AuditorAware<String> auditorProvider() {
+
+        return () -> Optional.ofNullable(((UserDetails) SecurityContextHolder.getContext()
```

Comment:
Are you sure, that NullPointerException will never be thrown here?

Reply:
yeeas, because this option only for an authorized user, corrected this